tasks/install-from-source.yml

```diff
@@ -1,27 +1,41 @@
 ---
+- name: Ensure dependency list is set (RedHat)
+  set_fact:
+    git_package_deps:
```

Comment:
I'd rather use the convention I have set up in some of my other roles (example: https://github.com/geerlingguy/ansible-role-mysql/blob/master/vars/Debian.yml), so instead of this, you'd have:

  1. A `vars/[OS-family].yml` file for each supported OS family (Debian.yml, RedHat.yml, Fedora.yml, etc.)
  2. A task that includes the variable variables file: https://github.com/geerlingguy/ansible-role-apache/blob/master/tasks/main.yml#L3-L4
  3. A task that defines the final new variable name if it's not already defined, using the `__` prefixed variable: https://github.com/geerlingguy/ansible-role-apache/blob/master/tasks/main.yml#L10-L13

Also, can you change the variable name to `git_install_from_source_dependencies`?